tests/test_utils.py

```diff
@@ -47,3 +47,9 @@ def test_continuous_quantile_from_discrete():
     y = np.array([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10] * 2)
     quantile_value = utils.continuous_quantile_from_discrete(y, 0.35)
     assert quantile_value == 3.5
+
+
+def test_convergence_parameters():
```

Comment:
IMO, numerical parameters for optimization (in this case, `loss_change` and `delta`) are better stored together in a class since they are similar concepts. As a result, it is easier to see if someone - inadvertently or maliciously - changed the parameters in `ConvergenceParameters` if there is a unit test for it.